tests/api/test_addon_uploads.py

```diff
@@ -1922,6 +1953,285 @@ def test_edit_version_both_license_and_custom_license(base_url, session_auth):
     ), f'Actual response message was {edit_version.text}'
 
 
+@pytest.mark.parametrize(
+    'value',
+    [
+        '',
+        'all-rights-reserved',
+        123,
+        None,
+        False,
+    ],
+    ids=[
+        'Empty string',
+        'Built in license',
+        'Integer',
+        'None/No value',
+        'Boolean',
+    ],
+)
+@pytest.mark.serial
+@pytest.mark.create_session('api_user')
+def test_edit_version_invalid_custom_license_format(base_url, session_auth, value):
+    """Custom licenses should be a dictionary containing the license name and text; other formats should fail"""
+    addon = payloads.edit_addon_details['slug']
+    request = requests.get(
+        url=f'{base_url}{_addon_create}{addon}',
+        headers={'Authorization': f'Session {session_auth}'},
+    )
+    # get the version id of the version we want to edit
+    version = request.json()['current_version']['id']
+    payload = {**payloads.custom_license, 'custom_license': value}
+    edit_version = requests.patch(
+        url=f'{base_url}{_addon_create}{addon}/versions/{version}/',
+        headers={
+            'Authorization': f'Session {session_auth}',
+            'Content-Type': 'application/json',
+        },
+        data=json.dumps(payload),
+    )
+    print(
+        f'For custom_license "{value}": Response status is '
+        f'{edit_version.status_code}; {edit_version.text}\n'
```

Comment:
Is this supposed to be here?

Reply:
I left the print there because it helps me debug failures more easily by checking the response code and the response message before the actual validations (assertions). 
I could add all of this in the first assert message (the one immediately following) as an alternative solution.